modules/data-entry/src/main/frontend/src/questionnaire/ComputedQuestion.jsx

```diff
@@ -0,0 +1,157 @@
+//
+//  Licensed to the Apache Software Foundation (ASF) under one
+//  or more contributor license agreements.  See the NOTICE file
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the
+//  "License"); you may not use this file except in compliance
+//  with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+
+import React, { useState } from "react";
+import PropTypes from 'prop-types';
+import { TextField, Typography, withStyles } from "@material-ui/core";
+
+import Answer from "./Answer";
+import AnswerComponentManager from "./AnswerComponentManager";
+import Question from "./Question";
+import QuestionnaireStyle from './QuestionnaireStyle';
+import { useFormReaderContext } from "./FormContext";
+
+let ComputedQuestion = (props) => {
+  const { existingAnswer, classes, ...rest} = props;
+  const { text, expression } = {...props.questionDefinition, ...props};
+  const [error, changeError] = useState(false);
+  const [errorMessage, changeErrorMessage] = useState(false);
+
+  let initialValue = existingAnswer && existingAnswer[1].value || "";
```

Comment:
Might be more tersely written as
```suggestion
  let initialValue = existingAnswer?.[1].value || "";
```